glide-core/src/client/value_conversion.rs

```diff
@@ -411,10 +416,43 @@ fn convert_array_elements(
     Ok(Value::Array(converted_array))
 }
 
-fn convert_array_to_map(
+/// Converts key-value elements in a given map using the specified types.
+///
+/// `map` A vector of key-values.
+/// `key_type` is used to convert each key when collecting into the resulting map.
+/// If `None` is given, then the key is not converted.
+/// `value_type` is used to convert each value when collecting into the resulting map.
+/// If `None` is given, then the value is not converted.
+fn convert_inner_map_by_type(
```

Comment:
isn't it the same as you did in #1489 in convert_map_entries? please reuse it

Reply:
It is similar.  Although this PR is much more general since it applies to, and can override all existing Map conversions.